File: src/main/java/top/criwits/scaleda/verilog/psi/factory/nodes/ModuleInstantiationPsiNode.kt
//package top.criwits.scaleda.verilog.psi.factory.nodes
//
//import com.intellij.lang.ASTNode
//import com.intellij.openapi.util.TextRange
//import com.intellij.psi.PsiReference
//import com.intellij.psi.util.PsiTreeUtil
//import org.antlr.intellij.adaptor.psi.ANTLRPsiNode
//
//class ModuleInstantiationPsiNodeOld(node: ASTNode) :
//    ANTLRPsiNode(node), ReferenceHolder<ModuleIdentifierPsiNode> {
//
//    override fun getHoldPsiNode(): ModuleIdentifierPsiNode? {
//        return PsiTreeUtil
//            .findChildOfType(this, ModuleIdentifierPsiNode::class.java)
//    }
//
//    override fun getHoldPsiNodeRelativeTextRange(): TextRange? {
//        return getHoldPsiNode()
//            ?.textRange
//            ?.shiftLeft(this.textOffset)
//    }
//
//    override fun getReference(): PsiReference {
//        return ModuleInstantiationReference(this)
//    }
//}

File: src/main/scala/top/criwits/scaleda/idea/verilog/references/ModuleInstantiationPsiNode.scala
package top.criwits.scaleda
package idea.verilog.references

import verilog.psi.factory.nodes.{ModuleIdentifierPsiNode, ReferenceHolder}

import com.intellij.lang.ASTNode
import com.intellij.openapi.util.TextRange
import com.intellij.psi.util.PsiTreeUtil
import org.antlr.intellij.adaptor.psi.ANTLRPsiNode

class ModuleInstantiationPsiNode(node: ASTNode)
  extends ANTLRPsiNode(node) with ReferenceHolder[ModuleIdentifierPsiNode] {
  override def getHoldPsiNode: ModuleIdentifierPsiNode =
    PsiTreeUtil.findChildOfType(this, classOf[ModuleIdentifierPsiNode])

  override def getHoldPsiNodeRelativeTextRange: TextRange =
    getHoldPsiNode.getTextRange.shiftLeft(this.getTextOffset)

  override def getReference = new ModuleInstantiationReference(this)
}
